csrc/host_ir/container.h

```diff
@@ -41,10 +42,19 @@ class HostIrContainer final : public Fusion {
     return top_level_exprs_.push_back(expr);
   }
 
+  void pushBackKernelExecutor(KernelExecutor* ke) {
+    return kernel_executors_.push_back(ke);
+  }
+
+  KernelExecutor* getKernelExecutor(int index) {
+    return kernel_executors_[index];
+  }
+
   Stream* getDefaultStream();
 
  private:
   std::vector<Expr*> top_level_exprs_;
+  std::vector<KernelExecutor*> kernel_executors_;
```

Comment:
I'd like HostIrContainer to own these KernelExecutors as said [here](https://docs.google.com/document/d/1QrRmN27XsVjZu7QrZWJJyRENO50878LC3MvlQY1cRYA/edit?tab=t.0#heading=h.ol33v89o9xl). This way, HostIrContainer can be executed standalone without relying on the data structure that actually owns the KernelExecutors. 

So I'd probably do `std::vector<std::unique_ptr<KernelExecutor>>` instead.